pkg/endpoint/endpoint.go

```diff
@@ -1057,6 +1064,42 @@ func (e *Endpoint) SetK8sNamespace(name string) {
 	e.unlock()
 }
 
+// SetNamedPorts sets the k8s container ports specified by kubernetes.
+// Note that once put in place, the new k8sPorts is never changed,
+// so that the map can be used concurrently without keeping locks.
+// Reading the 'e.k8sPorts' member (the "map pointer") *itself* requires the endpoint lock!
+// Can't really error out as that might break backwards compatibility.
+func (e *Endpoint) SetNamedPorts(containerPorts []types.ContainerPort) error {
+	k8sPorts := make(policy.NamedPortsMap, len(containerPorts))
+	for _, cp := range containerPorts {
+		if cp.Name == "" {
+			continue // skip unnamed ports
+		}
+		if !api.IsNamedPort(cp.Name) {
+			log.WithField(logfields.PortName, cp.Name).Warning("ContainerPort: Invalid port name, not using as a named port")
+			continue
+		}
+		name := strings.ToLower(cp.Name)
```

Comment:
I assume named ports are case insensitive?

Reply:
Yes. They are defined as `IANA_SVC_NAME`, which is defined as case insensitive in 
https://tools.ietf.org/rfc/rfc6335.txt